src/macros.jl

```diff
@@ -992,17 +993,52 @@ macro addNLConstraint(m, x, extra...)
     return assert_validmodel(m, code)
 end
 
-macro defNLExpr(x, extra...)
-    # Two formats:
-    # - @defNLExpr(a*x <= 5)
-    # - @defNLExpr(myref[a=1:5], sin(x^a))
-    length(extra) > 1 && error("in @defNLExpr: too many arguments.")
-    # Canonicalize the arguments
-    c = length(extra) == 1 ? x        : nothing
-    x = length(extra) == 1 ? extra[1] : x
+macro defNLExpr(args...)
+    if length(args) == 2
+        s = IOBuffer()
+        print(s,args[1])
+        print(s,",")
+        print(s,args[2])
+        msg = """
+        in @defNLExpr($(takebuf_string(s))): three arguments are required.
+        Note that the syntax of @defNLExpr has recently changed:
+        The first argument should be the model to which the expression is attached.
+        The second is the name of the expression (or collection of expressions).
+        The third is the expression itself.
+        Example:
+        @defNLExpr(m, my_expr, x^2/y)
+        @defNLExpr(m, my_expr_collection[i=1:2],sin(z[i])^2)
```

Comment:
Pedantic, but a space after the final comma would improve legibility so that it is clear that there are 3 args